spec/services/contacts_service_spec.rb

```diff
@@ -48,6 +49,22 @@
     end
   end
 
+  describe :upsert do
+    it 'raises a TypeError if filters is nil' do
+      expect(client.contacts.upsert(nil, { name: 'unique_name' }).to raise_error(TypeError)
+    end
+
+    it 'raises an ArgumentError if filters is empty' do
+      expect(client.contacts.upsert({}, { name: 'unique_name' }).to raise_error(ArgumentError)
+    end
+
+    it 'calls the upsert route with encoded filters' do
+      filters = { last_name: 'unique_name', 'custom_fields[external_id]': 'unique-1' }
+      contact = create(:contact, last_name: 'unique_name', custom_fields: { external_id: 'unique-1'})
+      expect(client.contacts.upsert(filters, contact)).to be_instance_of BaseCRM::Contact
+    end
```

Comment:
🤔 This test seems to cover the "update" path, but we are missing the "insert" path.
If the endpoint does an _upsert_ both scenarios should be tested

Reply:
Agreed, waiting on some guidance on the test suite in general here https://github.com/zendesk/basecrm-ruby/pull/83#discussion_r792602082 before i improve this but i agree